specs/waku/waku.md

```diff
@@ -519,6 +520,20 @@ Known static nodes MAY also be used.
 
 - Clarify the preferred way of handling unknown keys in the `status-options` association list.
 
+### Version 1
+
+Released []()
+
+- Removed `version` from handshake
+- Upgraded to `waku/1`
+
+### Version 0.5
+
+Released []()
+
+- Marked as [deprecated](./waku-0.md)
+- Corrected RLP keys to their int values
```

Comment:
This is not a specification change but mistake in implementation. Thus it should not be mentioned here imo.

Reply:
Although I agree with you that it should not be there and that's correct, pragmatically we had to make a change to the specs (RLP `0` became `48`) a posteriori.

Essentially the way I see it is that we decided that the specs in this case should reflect the live implementation (which is not a "correct" choice, but it's a pragmatic one as we only have one live implementation), so I feel it's useful to justify the change.

Having said that I don't feel to strongly about it, so happy to remove it if you don't buy the argument :)

Reply:
How about:

"Correct spec/implementation mismatch: Change RLP keys to be the their int values in order to reflect production behaviour"?

Reply:
Oh, I hadn't even noticed that the integers were changed in the ABNF specification. Which shows that this flow of doing the duplication of the file and making changes in it in one go (in one PR) is not good. I'm basically forced to do a diff between files locally on my machine to spot all changes.

That being said, I don't agree with changing the values. We can work around the problem just in the status-go implementation (accepting both integers), no? And this way remove it later again. Instead of uglifying the spec for the rest of its lifetime.

Reply:
OK, nevermind on changing the values. I see they were changed back in latest spec. Damn, confusing.
Can we please not do things like this because the implementation did something wrong?

Reply:
Anyway, we can leave it like this if it must. No other client will claim to have been supporting version 0.5.
Just, it is not necessary to be there and it feels like a weird thing to do. I wouldn't want this to become a habit in the future.

Reply:
> Which shows that this flow of doing the duplication of the file and making changes in it in one go (in one PR) is not good. I'm basically forced to do a diff between files locally on my machine to spot all changes.

That's fair, two PRs would have been a better way, I will split them.

> That being said, I don't agree with changing the values. We can work around the problem just in the status-go implementation (accepting both integers), no?

We can, it's more work, that means:
1) Make the handshake compatible between current botched version and actual v0 specs (which we will never use btw, as nimbus is not ready for prime time and status-go will jump to waku/1 directly likely)

2) Push out V1 removing the unused `version` field

As opposed to just 2) , specs might look not as "good", but is that really a valid point? developer working hours + complexity of code vs specs not looking aesthetically pleasing?

> Can we please not do things like this because the implementation did something wrong?

Not sure, if you have a better way to handle it I am all for it, as I said, this is clearly not the correct way, but pragmatically it's the one that will lend itself to the least amount of complexity  in the currently only live implementation and saves some dev hours.

Effectively whoever will implement `waku/0` now will be able to talk to the only currently live implementation, and once we push out `waku/1` similarly it should, which I think it's what matters most.

Reply:
I still don't really understand why the spec needs the inbetween 0.5 update. You can just have a version 0 in status-go that is not compliant to the waku version 0 specification. At least, if then you are going to update it to version 1. I don't see the practical difference and thus don't understand why you say it is pragmatically the way to handle it, except for being correct according to the letter of the spec, that is by altering the spec.

But, as version 0.5 will not really be used and is just there for backwards compatibility in status-go, it is no big deal for me. No value in wasting more man hours discussing it :-).

Main issue here was the confusion created by doing the two version releases in one PR.

Reply:
I agree that there's not value in discussing this any further, either ways I don't think it makes a difference, I am ok with leaving it as it or changing it, so I will leave it as it unless someone as a strong opinion.